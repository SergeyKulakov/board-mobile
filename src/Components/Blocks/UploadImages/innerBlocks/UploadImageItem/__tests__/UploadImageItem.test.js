import React from 'react'
import { shallow } from 'enzyme'

import 'jest-styled-components'

import { UploadImageItem as Component } from '../index'

describe('UploadImageItem', () => {
  const defaultProps = {
    userId: 'userId',
    id: '1234',
    image: 'image',
    type: 'type',
    link: 'link',
    isDeletable: true,
    jobImagesMode: 'jobImagesMode',
    onDelete: () => null,
    onItemClick: () => null,
    onSetLinkData: () => null,
    resizeMode: 'contain',
  }

  it('render snapshot', () => {
    const wrapper = shallow(<Component {...defaultProps} />)
    const wrapper2 = shallow(<Component {...defaultProps} image={null} />)

    expect([wrapper, wrapper2]).toMatchSnapshot()
  })

  it('update image', () => {
    const props = {
      ...defaultProps,
      jobImagesMode: null,
    }
    const wrapper = shallow(<Component {...props} />)

    expect(wrapper.state('image')).toBe('image')

    wrapper.setProps({
      image: 'image2',
    })

    expect(wrapper.state('image')).toBe('image2')
  })
})
