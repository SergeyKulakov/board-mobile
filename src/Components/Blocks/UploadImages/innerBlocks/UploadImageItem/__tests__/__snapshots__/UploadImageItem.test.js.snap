// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadImageItem render snapshot 1`] = `
Array [
  <TouchableOpacity
    activeOpacity={0.2}
    disabled={false}
    onPress={[Function]}
  >
    <Styled(ImageBackground)
      resizeMode="contain"
      source="image"
    >
      <Styled(TouchableOpacity)
        hitSlop={
          Object {
            "bottom": 5,
            "left": 5,
            "right": 5,
            "top": 5,
          }
        }
        onPress={[Function]}
      >
        <Icon
          color="#fff"
          name="times"
          size={12}
        />
      </Styled(TouchableOpacity)>
    </Styled(ImageBackground)>
  </TouchableOpacity>,
  <TouchableOpacity
    activeOpacity={0.2}
    disabled={false}
    onPress={[Function]}
  >
    <ActivityIndicator
      animating={true}
      color="#999999"
      hidesWhenStopped={true}
      size="small"
    />
    <Styled(TouchableOpacity)
      hitSlop={
        Object {
          "bottom": 5,
          "left": 5,
          "right": 5,
          "top": 5,
        }
      }
      onPress={[Function]}
    >
      <Icon
        color="#fff"
        name="times"
        size={12}
      />
    </Styled(TouchableOpacity)>
  </TouchableOpacity>,
]
`;
